### DISCUSSION TITLE
USB functionality after firmware update

### DISCUSSION THREAD

**Discussion by fedordragan** [Desuuuu/Marlin#23] [General]
Hi there Desuuuu! I have a CR10S Pro V2, and wanted to update stock firmware to yours. I compiled and flashed marlin to 2.0.7.2, and then flashed the screen. I followed the wiki https://github.com/Desuuuu/DGUS-reloaded/wiki/Flashing-the-firmware

Downloaded Marlin source from here https://github.com/Desuuuu/Marlin/releases/tag/2.0.7.2, 
Changed config files to the CR10SProV2 flavor from https://github.com/Desuuuu/DGUS-reloaded-config/tree/2.0.7.2/examples/Creality/CR10SProV2 - used unchanged from yours but I'm attaching here just in case
Build with PlatformIO, all vent well. Flashed resulting hex with Xloader (115200 baud)

I've got the DWIN_SET from https://github.com/Desuuuu/DGUS-reloaded/releases/tag/1.0.2, however I also added there files from extras - for flash reset and core firmware https://github.com/Desuuuu/DGUS-reloaded/tree/master/extras

All went well, all the functionality of the touchscreen is working, including custom G-code. INSTEAD, now I can't send any gcode via Pronterface (or Cura, etc). As soon as I connect it gives basic info and that's it, no response to M105. A command sent from console gives an error such as 
>>> M503
[ERROR] Printer is not online.

Strangely enough, at the same time, I see echoes of all the commands that I send via touchscreen - see in the log attached there was a M503 sent from the screen. Pronterface communicates at baud 250000. I can still re-flash firmware via Xloader, however only using 115200 baud, strangely enough it gives an error if I try 250000. 

Any suggestions of anything else I can try?

Thanks,
Fedor

[config.zip](https://github.com/Desuuuu/DGUS-reloaded/files/5857688/config.zip)
[Pronterface log.txt](https://github.com/Desuuuu/DGUS-reloaded/files/5857423/Pronterface.log.txt)
![Pronterface](https://user-images.githubusercontent.com/17398012/105522732-2afe9f80-5cb8-11eb-961e-03bc7334ba83.png)

**Comment by Desuuuu** [comment]
I'd suggest starting by changing the baudrate in `Configuration.h` to 115200, re-compiling and re-flashing.

If that doesn't work, you might wanna try an older release (both the firmware and configuration repos have corresponding tags for each version).

**Reply by fedordragan** [reply]
Changing baud to 115200 didn't help. 
Flashed Marlin 2.0.6.1 and everything working fine now, thank you!

**Reply by Desuuuu** [reply]
That's interesting. I haven't seen an issue upstream related to that problem so I wonder if there's an issue with the configurations.

Even though I have an SKR board in my printer, I tried to replicate this using a relatively similar configuration (where it should matter at least) but I haven't been able to. It might be because of the different HAL I guess.

Unfortunately, I most likely won't be able to troubleshoot this.

**Reply by fedordragan** [reply]
Yeah... I think in a recent thread another user had similar issues https://github.com/Desuuuu/Marlin/discussions/22

At least we know that 2.0.6.1 works. I can try out any bug fix or future release on my machine and report back whether the issue is still there.

**Reply by fedordragan** [reply]
So I noticed today that a USB stopped working again - this time on 2.0.6.1. The only thing that I did since is to try a PID autotuning from screen. I reflashed 2.0.6.1 but that didn't work. Then I flashed 2.0.7.2 and then USB started working again. 
I've tried to replicate my actions, pressed PID autotune command and after it was complete USB stopped working again lol :) 

What's confusing is that flashing the same version wouldn't help. In order to get back to a working USB on 7.2, I had to flash .6.2 and then 7.2 again. 

So... could it be related to that? I should also mention that I see some weird symbols in the corner whenever I send a command, see attached. 
![image](https://user-images.githubusercontent.com/17398012/105690723-e6048400-5ed6-11eb-98ae-54bed0211e3e.png)

**Comment by Desuuuu** [comment]
Thanks for the details, I'll look into it!

Just to be sure, does this happen if you run PID autotuning (with [M303](https://marlinfw.org/docs/gcode/M303.html)) from serial?

**Reply by fedordragan** [reply]
Yes, same thing. Here's what I got on the terminal: 
>>> M303 E0 S200 U1
SENDING:M303 E0 S200 U1
**Attempted to write invalid text to console, which could be due to an invalid baudrate
Attempted to write invalid text to console, which could be due to an invalid baudrate
!z!>**
PID Autotune start
 bias: 140 d: 114 min: 197.07 max: 202.78
 bias: 139 d: 115 min: 197.73 max: 202.78 Ku: 57.98 Tu: 18.02
 Classic PID
 Kp: 34.79 Ki: 3.86 Kd: 78.38

USB continued to work after the task was finished.

**Reply by Desuuuu** [reply]
Thanks for the log. It's pretty interesting that there's an error message in the middle of the output.

**Reply by fedordragan** [reply]
Here's the message from octoprint:
Send: M303 E0 S200 U1
Recv: //action:notification ÔåÜyã3m¶q\x8eÞyÓ¬¦÷Óí¦\x97hn®\xff\x1dmó#\xff\x9cÔâÍÍó³_þ\x96þÝ{þÀOîý½47á\x8ec¶\x8ffv?\x97W-\x8dÈmNV´¿oTzÇþoûýó7½\x87Ín7ß]ôÝoöÛü³¶]þ7î÷
¾÷^}2îkû[}\x9eõ\x97÷N{ç\x83\x8aY\xff÷MÏª{p\x15=ÓG~\x1fè \x8eÝ\x99Ù8ÐG\x1e\x7f;¢L\x1aÙ7\x8d~\x9b\x93Î;¿ó\x17¢·§Ý\x15\x1b®O¾kôÙÂÕ¸û¿>\x1eôj\x08Ù÷ï»ón½uAÑ¸Æ_ÛïìýãÖ\xff\»ï\xffIUoS_²\x9aýþäjÊL\xffê¾ìÜö¼Óµ\xffþë\xffÄ\x07ô(m\x8fý×Ú8ôú½þënÆ}ïù¾þ²xÏ¶{¹\xffçxÐMîÖñi\x90\x1ekÖ;Ù§8õ¹üº\x96%.?ËBS&õñä[Ôýu\x9e\xff\x92,D\x9fW÷>9\xffé\xffü^ën?ß÷W[÷znqù¾¹ºþ\x99ìÛ®{þl\x19\x88¿Ê;¾³Þ\x9f\x9f·wük_(ï/ßïþ2LÜæûÑx.n\x8b\xff¾\x9d\xff\x1f±=\xff¹ú÷\x93º)½\x1a~®"ß\x17W¿Í®æ\x80î\x1ch[úý?Æ}:þw\x97·Ý}¼yö\x7f¦ÏÓôù¬±é¯GößÞ\x1c\x0b\x9e\xff¦ûJ\x02Íë¶O¡iî»¼ýçò§Åókí\xff¢Æ\x99y\x9fvÛèÞ\x7fë\x96¼·ïé3Ú\x9bnF­ï|>\x9dÍúãÇ\x86\x0b(·mó»¾\xff\x08¾çeý9³\x9cê4°øÎÉÞø\x0fþjÚÜ+äÆ¬\½îö¯\x8fóä|]äøûûýW\x10\x8b÷ï?oÎ{¿è_[îþ\x8a&ºø½â¾Q5¾¬}ýï\x9a\x93\x9c14ý&ü\x14ö¼%\x9e\xff¯\x9e\x9dãþÝf¿{Å<ó\x7f\xffÝÏNõ\x1ai:\x8dZ»nå¢ÊÑûý\\x9fÏ\x01'\çÏ?\x97Ï¢\x17·\x9ea÷×\xffA¹\x02\x9e~~o\x15h^;§®'ç¥\xff\x8f\x9b¥.møjÞ¢ºeò9¿üzó¹f¦)v°ê9o¥\x83hû½ùºI\xff­+{\xffÙôÕo8lýü\x7fû½:Í\x7fvíÏ¸ï,ùjÛZSy÷:\x91ýê[ùê×ØZF{õkøùæ:Ö>ùî÷\ÊÛÜ\x8d\x0fs­\x05î'x_ý\x0f«ïøýÞ<É5÷ëÅò\x95×´íýsôï.}\x7f®Û¸ò\x8f;_zÇå½'¯\x07×\xff§\xffáþX-Ñ\x7fm;\x93î|ßñk¹ú±\xff\x9fÜ©«\x9fõI»ÛY{_Î\x8f\x7fM¬9Þó\xffÑßîÏ\x7f\xff@·öue;\x0c3ß\x988wjØ'
Recv: Ø]wù}\x1a­\x8b­a?®\x9cþê÷½¸\x9f\x8a® ü}Ï®¶~ÖÎö\x8e\x1fëÌnözG}óÍã"ëý,¹¬¦\xff¢ú÷âñ\x1e\x07/û\x9aîîù\xff(gdSf÷¡~ã\x93jîIõ|¢(×i¿\x9fõo\x87Ü"µïñ­+¦îº³\xffÕÜÈ¶;â³Û«ó\x7fß¢·ÏUQäëõ^ù\x1bÓ÷¸\x7fþì÷£Ø[?ýKÃÕþ÷ýN»ç­ÓWo\xffwÆþW{7\x86RY³Ý?\x17ü\xffÖ\x9dØu\yN¿ZN¬îÙ\x9d­t×_ýGÖ2\xffw\x7fAÚ\x1ePÁ5\x8b\x9d\x8bßñw\x8e<9½ê\x0f1\x9dWø]/\x7f¶9\x7f~ªæ\x9f­8\x95ó¼þ2Ó_Õ\xffÓõÜ5¾Û¾Ç<\x05¸\x1b\x9c{·øÏ=ßC=tÛóU^g\x9b.æ¿ûvj\x9eë\x7fÎ}zÅÓªÉ;\xffß~\x87êJ\x1aýºß\x9aã\x95®økwM]:\x93¯\x9e¯\x7fß\x80\x98vG^××ÓïÜæWþa-\x99Óõ2\x1fAº®í\x7fã»ó\x9boªCßwø×@ûß®C÷éË÷­þ\x7fÝ¬ùà«&>T·½oùUzÖ®\x9fvÊ\x0c­\xff\xffæ-º\x0eÏ²»£§K]\x19\x7fþ \x7f\xff\x84\x9dØõº¹Ôç\x8b½\x13\x1d{;ûÑõMc\x8d|î½ïfï¾÷ñ~¦ó÷Í³ß¹[[\xffu¼×¯ów/Îýçýo\x1e^ù:þÝõ/ï_Þc×b¿\xffûä\x81\x9f¾Ã|ÉüÞ{?þ7úÖ\x1f=Fþå\x95W×Ü\x8f|í÷g\xff2þ\x8d?UÛÏÎyû\x0e\x03&ÛûnÏ~¿¾á}~¿\xffl¸ûý[j\xffï/\x7fÍû\x1c'³\x95kký×üè\xff\x7fçk~\x1dÛ^\x9eÝÏûn\x86W^ñ\x1d¹Þ¹¦O\x7f\x9bü\x92zÔ{Ï¹×çO·\x97\xff\x18¿â ºÏ×³ÚmÞá^õxF¾\x9eO·eú3«øG7¾y_s_z7ÒöHï\x95%Ò~i\x7fboùë_ÎN\x8e¿èý\x9fû¿ãr¿mï\x8f\x93µÙ\x1f^»ïò÷ãC>*»éýoôÚ\x9f\x86ç[þ¹­®\x1eç\x1fÎ=Iý7Õ\xff{zûïýëçµätÎ\xffêwÝ\x16÷O|.Lå÷ñw¿ï5Æ?ßý½Þóý«ÞwîYZ·\x7fßtÖþ«æÜ{ËúÝ­.äþkò¶7ûß\x1fï{æÜ\x8a\x0b\x9f}\x94¸\x0f\x9f~=\xff\x9fß\x93\x93×Ônî\xff\x17Ý\x1f|o÷\x7f,÷_?¸Æm6»Aúü\xff¤_;\x89mº\x9aõýüÇ\x97ÅÕ={\x1aåÚ[õ×ß«µò1ï£Í§\xff4e7jgn\x9f\x9eË§óíó5ã«ýTýå\x7f~ð§·ûo\xff_ù\x91O.õª\x97\x7f\x98Å\x1f¼{n_äwöÝ&ìI»ïwÖÓ\xffæÏï`<\x93ÓçÈlvt¹ñó½¶OÒ\x1eþþû\x82ûÜß6\x96üÙë0oýÛ\xff®ßÜ¿ý»ék®_ß\x87\x9cÜìr×Ú\x9bîÏ«ÝÐ\
Recv: PID Autotune start
[...]

**Reply by Desuuuu** [reply]
Thanks, I was able to locate the issue.

It's a problem in Marlin's code. I'll submit a pull request to their repository to fix it.

Edit: It's already fixed in Marlin's bugfix branch since Nov. 13th 2020 so it will be included in the next release.

**Comment by Desuuuu** [comment]
Considering the slow release cycle of Marlin, I just applied my changes to the latest `bugfix-2.0.x` branch and pushed them to the [bugfix](https://github.com/Desuuuu/Marlin/tree/bugfix) branch of my repository. The problem should be fixed there if you want to give it a try.

It should compile fine but I haven't tested the firmware. Please note I don't plan on updating this branch very regularly and I won't be providing binaries/configurations for it.

**Reply by DaKrampus** [reply]
OK I had the same problem with pronterface... i tried the bugfix version... took me a good hour to find out that version didnt like my older config file.. but i managed to compile and...... it works.. i am able to login with pronterface again.. (i am on a cr10S pro with bltouch added..)
congrats it seems to work perfectly.. 
keep up the good work and thank you for the update!

Luc
